SPARCore/GADArticle.m

```diff
@@ -2,6 +2,97 @@
+                                                  *_Nullable articles,
+                                                  NSError *_Nullable error))completion {
+    NSURL *queryURL = [self urlForArticlesFromPublication:publicationId];
+    
+    [super fetchModelsWithParams:queryURL
+                          queryParameters:@{}
+                         modelTransformer:^(NSData *jsonData) {
+                            return [self articlesFromJSON:jsonData];
+                         }
+                        completionHandler:completion];
+
+}
+
+- (void) fetchFulltextWithCompletion: (void(^_Nonnull)(GADRemoteModel * *_Nullable model,
+                                                NSError *_Nullable error))completion {
+    NSURL *queryURL = [GADArticle createURLWithArticle:self.articleId publication:self.publicationId];
+    
+    [GADRemoteModel fetchModelsWithParams:queryURL
+                          queryParameters:@{}
+                         modelTransformer:^(NSData *jsonData){
+                             return [GADArticle articlesFromJSON:jsonData];
+                         }
+                        completionHandler:^(NSArray <GADArticle *> *_Nullable model, NSError *_Nullable error){
+                            GADArticle *fullArticle = model[0];
+                            self.content = fullArticle.content;
+                            self.authors = fullArticle.authors;
+                            //completion(); --How should we call the completion handler here?
+                        }];
+    
+}
+
++ (NSArray <GADArticle *> *) articlesFromJSON: (NSData *)json {
+    
+    NSMutableArray <GADArticle *> *articles = [[NSMutableArray alloc] init];
+    NSError *JSONParsingError;
+    NSArray *jsonArray = [NSJSONSerialization JSONObjectWithData:json
+                                                         options:kNilOptions error:&JSONParsingError];
+    
+    for (NSDictionary *element in jsonArray) {
+        GADArticle *article = [self articleFromDictionary:element];
+        [articles addObject:article];
+    }
+    return articles;
+}
+
++ (GADArticle *)articleFromDictionary:(NSDictionary*)dict {
+    //Map fields of element to fields of article
+    GADArticle *article = [[GADArticle alloc] init];
+    //datePublished field is a UNIX Timestamp number - converting to NSDate here
+    int timeStamp = (int)dict[API_DATE_PUBLISHED];
+    article.datePublished = [NSDate dateWithTimeIntervalSince1970: timeStamp];
+    article.brief = dict[API_BRIEF];
+    article.headerImage = dict[API_HEADER_IMAGE];
+    article.publicationId = dict[API_PUBLICATION_ID];
+    article.articleId = dict[API_ARTICLE_ID];
+    article.title = dict[API_TITLE];
+    if (dict[API_CONTENT]) {
+        article.content = dict[API_CONTENT];
+        article.authors = dict[API_AUTHORS];
+    }
+    return article;
+}
+
++ (NSURL *) urlForArticlesFromPublication: (NSString *)publicationId{
```

Comment:
Maybe this should be a method on `GADPublication` @arthurhero? `- (NSURL *)articlesURL` wouldn't require the `publicationId` (because then we could use `[self publicationId]`).